app/gui2/src/util/ast/__tests__/parsingAPI.test.ts

```diff
@@ -0,0 +1,326 @@
+              },
+              type: 'Number',
+              whitespaceLengthInCodeParsed: 1,
+              whitespaceStartInCodeParsed: 18,
+            },
+            type: 'OprApp',
+            whitespaceLengthInCodeParsed: 0,
+            whitespaceStartInCodeParsed: 15,
+          },
+          newline: {
+            lengthInCodeBuffer: 1,
+            startInCodeBuffer: 14,
+            type: 'Newline',
+            whitespaceLengthInCodeBuffer: 0,
+            whitespaceStartInCodeBuffer: 14,
+          },
+        },
+      ],
+      type: 'BodyBlock',
+      whitespaceLengthInCodeParsed: 0,
+      whitespaceStartInCodeParsed: 0,
+    },
+  },
+  {
+    code: 'function file=var',
+    expected: {
+      childrenLengthInCodeParsed: 17,
+      statements: [
+        {
+          expression: {
+            arg: {
+              childrenLengthInCodeParsed: 3,
+              token: {
+                lengthInCodeBuffer: 3,
+                liftLevel: 0,
+                startInCodeBuffer: 14,
+                type: 'Ident',
+                whitespaceLengthInCodeBuffer: 0,
+                whitespaceStartInCodeBuffer: 14,
+              },
+              type: 'Ident',
+              whitespaceLengthInCodeParsed: 0,
+              whitespaceStartInCodeParsed: 14,
+            },
+            childrenLengthInCodeParsed: 17,
+            equals: {
+              lengthInCodeBuffer: 1,
+              startInCodeBuffer: 13,
+              type: 'Operator',
+              whitespaceLengthInCodeBuffer: 0,
+              whitespaceStartInCodeBuffer: 13,
+            },
+            func: {
+              childrenLengthInCodeParsed: 8,
+              token: {
+                lengthInCodeBuffer: 8,
+                liftLevel: 0,
+                startInCodeBuffer: 0,
+                type: 'Ident',
+                whitespaceLengthInCodeBuffer: 0,
+                whitespaceStartInCodeBuffer: 0,
+              },
+              type: 'Ident',
+              whitespaceLengthInCodeParsed: 0,
+              whitespaceStartInCodeParsed: 0,
+            },
+            name: {
+              lengthInCodeBuffer: 4,
+              liftLevel: 0,
+              startInCodeBuffer: 9,
+              type: 'Ident',
+              whitespaceLengthInCodeBuffer: 1,
+              whitespaceStartInCodeBuffer: 8,
+            },
+            type: 'NamedApp',
+            whitespaceLengthInCodeParsed: 0,
+            whitespaceStartInCodeParsed: 0,
+          },
+          newline: {
+            lengthInCodeBuffer: 0,
+            startInCodeBuffer: 0,
+            type: 'Newline',
+            whitespaceLengthInCodeBuffer: 0,
+            whitespaceStartInCodeBuffer: 0,
+          },
+        },
+      ],
+      type: 'BodyBlock',
+      whitespaceLengthInCodeParsed: 0,
+      whitespaceStartInCodeParsed: 0,
+    },
+  },
+])('Parsing $code', ({ code, expected }) => {
+  expect(debug(parseEnso(code))).toMatchObject(expected)
+})
```

Comment:
This is likely a good candidate for a [snapshot test](https://vitest.dev/guide/snapshot.html#snapshot).

```suggestion
])('Parsing $code', (code) => {
  expect(debug(parseEnso(code))).toMatchSnapshot()
})
```